Katello/katello
# Fixes #36417 - Bring back 'Republish repository metadata'

#### What are the changes introduced in this pull request?

1. Add a 'Republish Repository Metadata' action to the repository details actions:
![image](https://github.com/Katello/katello/assets/22042343/b5ca7a6f-fbb4-4dad-943d-521e6d86dba8)
2. Add a 'Republish Repository Metadata' action to the Content view version details actions:
![image](https://github.com/Katello/katello/assets/22042343/f81602d2-4f60-47e8-aa5f-996d70f45864)
 
#### Considerations taken when implementing this change?

As we know, this action is dangerous for repositories using the 'Complete' mirroring policy, and the API requires the `force` option to be passed. With this in mind,

1. Any repositories that use the complete mirroring policy will have the action disabled.
2. For content view versions, the action will be disabled if _any_ of the content view version's repositories use complete mirroring.

You can still republish repo metadata in such cases; however, you'll have to use API or Hammer and not the web UI.

One caveat (described in more detail in https://bugzilla.redhat.com/show_bug.cgi?id=2184278#c8) is that this won't take into account whether a repository's mirroring policy has changed. We just have to live with that for now.

#### What are the testing steps for this pull request?

1. Verify that republishing repository metadata works as expected on the Repository detail page
2. Verify that republishing repository metadata works as expected on the Content view version details page
4. Verify that the option is disabled in both places when complete mirroring is set
